viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+
+    movie ={}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for item in user_data["watchlist"]:
+        if item["title"] == title:
+            user_data["watchlist"].remove(item)
+            user_data["watched"].append(item)
+    return user_data
+        
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    
+    list_of_ratings = [item["rating"] for item in user_data["watched"]]
+
+    # This will account for the situation when the watched list is empty
+    if len(list_of_ratings) == 0:
+        list_of_ratings.append(0)
+    
+    return sum(list_of_ratings)/len(list_of_ratings)
+
+def get_most_watched_genre(user_data):
+ 
+    list_of_genres = [item["genre"] for item in user_data["watched"]]
+
+    # This will account for the situation when the watched list is empty
+    if len(list_of_genres) == 0:
+        return None
+    
+    # Getting the mode of the list of genres
+    popular_genre = max(set(list_of_genres), key = list_of_genres.count)

```

Comment:
Clever use of max and set! As extra practice, can you consider how you could do this with a normal for loop instead of max and set?